src/Tools/AnalyzerRunner/IncrementalAnalyzerRunner.cs

```diff
@@ -50,26 +48,13 @@ internal async Task RunAsync(Workspace workspace, CancellationToken cancellation
 
             var exportProvider = (IMefHostExportProvider)workspace.Services.HostServices;
 
-            var solutionSizeTracker = exportProvider.GetExports<ISolutionSizeTracker>().Single().Value;
-
             // This will return the tracker, since it's a singleton.
-            var analyzer = ((IIncrementalAnalyzerProvider)solutionSizeTracker).CreateIncrementalAnalyzer(workspace);
-            await analyzer.NewSolutionSnapshotAsync(workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
 
             var solutionCrawlerRegistrationService = (SolutionCrawlerRegistrationService)workspace.Services.GetRequiredService<ISolutionCrawlerRegistrationService>();
             solutionCrawlerRegistrationService.Register(workspace);
-            solutionCrawlerRegistrationService.WaitUntilCompletion_ForTestingPurposesOnly(workspace, ImmutableArray.Create(analyzer));
-
-            var size = solutionSizeTracker.GetSolutionSize(workspace, workspace.CurrentSolution.Id);
-            Console.WriteLine("Current solution size:\t" + size);
 
             if (_options.UsePersistentStorage)
```

Comment:
📝 Currently this pull request breaks the `UsePersistentStorage` option. We either need to remove `NoOpPersistentStorage`, or modify this code to ensure that's the storage used when `UsePersistentStorage` is false.

Reply:
very true.

Reply:
Fixed.